extensions/ql-vscode/src/data-extensions-editor/model-details/model-details-panel.ts

```diff
@@ -24,14 +24,21 @@ export class ModelDetailsPanel extends DisposableObject {
   }
 
   public async setState(
-    externalApiUsages: ExternalApiUsage[],
-    databaseItem: DatabaseItem,
+    externalApiUsages?: ExternalApiUsage[],
+    databaseItem?: DatabaseItem,
+    hideModeledApis?: boolean,
   ): Promise<void> {
-    await this.dataProvider.setState(externalApiUsages, databaseItem);
-    this.treeView.badge = {
-      value: externalApiUsages.length,
-      tooltip: "Number of external APIs",
-    };
+    await this.dataProvider.setState(
+      externalApiUsages,
+      databaseItem,
+      hideModeledApis,
+    );
+    if (externalApiUsages) {
```

Comment:
I agree with your suggestion that checking against `undefined` is safer, but an empty array is a truthy value in JavaScript, so this would actually work.

Reply:
Oh sorry, I had in my head it was falsy. 🙇🏼

Reply:
👍 I'll use your suggestion nonetheless to limit confusion!